lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -85,16 +92,17 @@ def failure_message
     #   "app" => ["array of log lines", "received from app container"],
     #   "nginx" => ["array of log lines", "received from nginx container"]
     # }
-    def fetch_logs(kubectl)
+    def fetch_logs(kubectl, since: nil, timestamps: false)
       return {} unless exists? && @containers.present?
       @containers.each_with_object({}) do |container, container_logs|
         cmd = [
           "logs",
           @name,
           "--container=#{container.name}",
-          "--since-time=#{@deploy_started_at.to_datetime.rfc3339}",
+          "--since-time=#{(since || @deploy_started_at).to_datetime.rfc3339(9)}",
```

Comment:
While the server accepts fractional seconds, its still gives a .2-.3 seconds worth of overlap so I manually filter on line 156.